cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/upscale/AzureUpscaleService.java

```diff
@@ -114,22 +124,45 @@
             return ListUtils.union(Collections.singletonList(new CloudResourceStatus(armTemplate, ResourceStatus.IN_PROGRESS)),
                     successfulInstances);
         } catch (Retry.ActionFailedException e) {
+            LOGGER.error("Retry.ActionFailedException happened", e);
             rollbackResources(ac, client, stack, cloudContext, resources, preDeploymentTime);
             throw azureUtils.convertToCloudConnectorException(e.getCause(), "Stack upscale");
         } catch (CloudException e) {
+            LOGGER.error("CloudException happened", e);
             rollbackResources(ac, client, stack, cloudContext, resources, preDeploymentTime);
+            checkIfQuotaLimitIssued(e);
             throw azureUtils.convertToCloudConnectorException(e, "Stack upscale");
         } catch (RolledbackResourcesException e) {
+            LOGGER.error("RolledbackResourcesException happened", e);
             rollbackResources(ac, client, stack, cloudContext, resources, preDeploymentTime);
             throw new CloudConnectorException(String.format("Could not upscale Azure infrastructure, infrastructure was rolled back with resources: %s, %s",
                     stackName, e.getMessage()), e);
         } catch (Exception e) {
+            LOGGER.error("Exception happened", e);
             rollbackResources(ac, client, stack, cloudContext, resources, preDeploymentTime);
             throw new CloudConnectorException(String.format("Could not upscale Azure infrastructure, infrastructure was rolled back: %s, %s", stackName,
                     e.getMessage()), e);
         }
     }
 
+    private void checkIfQuotaLimitIssued(CloudException e) throws QuotaExceededException {
+        if (e.body() != null && e.body().details() != null) {
+            List<CloudError> errorDetails = e.body().details();
+            for (CloudError errorDetail : errorDetails) {
+                if ("QuotaExceeded".equals(errorDetail.code())) {
+                    Pattern pattern = Pattern.compile(".*Current Limit: ([0-9]+), Current Usage: ([0-9]+), Additional Required: ([0-9]+).*");
+                    Matcher matcher = pattern.matcher(errorDetail.message());
+                    if (matcher.find()) {
```

Comment:
Can you please add an else branch and log if it was a quota issue but the pattern didn't match? I assume in that case we can't do best effort scaling right?

Reply:
good idea, thanks!

Reply:
Yes, we don't do best effort if it does not match.